core/src/main/scala/akka/persistence/r2dbc/journal/JournalDao.scala

```diff
@@ -74,13 +74,27 @@ private[r2dbc] class JournalDao(journalSettings: R2dbcSettings, connectionFactor
 
   private val journalTable = journalSettings.journalTableWithSchema
 
-  // The subselect of the db_timestamp of previous seqNr for same pid is to ensure that db_timestamp is
-  // always increasing for a pid (time not going backwards).
-  // TODO we could skip the subselect when inserting seqNr 1 as a possible optimization
-  private val insertEventSql = s"INSERT INTO $journalTable " +
-    "(slice, entity_type, persistence_id, sequence_number, writer, adapter_manifest, event_ser_id, event_ser_manifest, event_payload, meta_ser_id, meta_ser_manifest, meta_payload, db_timestamp) " +
-    "VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, GREATEST(transaction_timestamp(), " +
-    s"(SELECT db_timestamp + '1 microsecond'::interval FROM $journalTable WHERE slice = $$13 AND entity_type = $$14 AND persistence_id = $$15 AND sequence_number = $$16)))"
+  private val (insertEventWithParameterTimestampSql, insertEventWithTransactionTimestampSql) = {
+    val baseSql =
+      s"INSERT INTO $journalTable " +
+      "(slice, entity_type, persistence_id, sequence_number, writer, adapter_manifest, event_ser_id, event_ser_manifest, event_payload, meta_ser_id, meta_ser_manifest, meta_payload, db_timestamp) " +
+      "VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, "
+
+    // The subselect of the db_timestamp of previous seqNr for same pid is to ensure that db_timestamp is
+    // always increasing for a pid (time not going backwards).
+    // TODO we could skip the subselect when inserting seqNr 1 as a possible optimization
+    def insertSubSelect(p: Int) =
+      s"(SELECT db_timestamp + '1 microsecond'::interval FROM $journalTable " +
+      s"WHERE slice = $$$p AND entity_type = $$${p + 1} AND persistence_id = $$${p + 2} AND sequence_number = $$${p + 3})"
+
+    val insertEventWithParameterTimestampSql =
+      baseSql + "GREATEST($13, " + insertSubSelect(p = 14) + "))"
+
+    val insertEventWithTransactionTimestampSql =
+      baseSql + "GREATEST(transaction_timestamp(), " + insertSubSelect(p = 13) + "))"
+
+    (insertEventWithParameterTimestampSql, insertEventWithTransactionTimestampSql)
```

Comment:
The shift of param bindings is annoying and obfuscate a bit the code intention. 
Maybe better if we just use out-of-order params? I'm not sure if keeping the ordering is buying use anything.

Reply:
yeah, it's annoying. I'd prefer to have clean sql strings in the end because those may show up in logs and such.
we could add some small utility that does some string replacement of `$` to add the numbered $1, $2, $3.

Reply:
created issue https://github.com/akka/akka-persistence-r2dbc/issues/135 for this